nilearn/_utils/data_gen.py

```diff
@@ -751,6 +750,25 @@ def basic_confounds(length, random_state=0):
     return confounds
 
 
+def add_metadata_to_bids_dataset(bids_path,
+                                     metadata,
+                                     json_file=None):
+    if json_file is None:
```

Comment:
if the function is meant to be part of the public API, it should have a docstring.
Otherwise the function name should have a leading underscore

Reply:
done